From 3b899c124e4753401e6bb4f1e71a249ac634c72e Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Thu, 29 May 2025 16:54:09 +0200
Subject: [PATCH] lxc: revert print function for corect function with turris
 repo

Revert print function in lxc-download file for correct function with turris
repository. Due to changes in print function, containers from our repository
not showing under LuCI, this revert fixes it. This changes doesn't affect
repository of images.linuxcontainers.org
---
 .../040-revert-back-names-turris.patch        | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)
 create mode 100644 utils/lxc/patches/040-revert-back-names-turris.patch

diff --git a/utils/lxc/patches/040-revert-back-names-turris.patch b/utils/lxc/patches/040-revert-back-names-turris.patch
new file mode 100644
index 000000000..dec96240c
--- /dev/null
+++ b/utils/lxc/patches/040-revert-back-names-turris.patch
@@ -0,0 +1,22 @@
+Index: lxc-6.0.4/templates/lxc-download.in
+===================================================================
+--- lxc-6.0.4.orig/templates/lxc-download.in
++++ lxc-6.0.4/templates/lxc-download.in
+@@ -232,7 +232,7 @@ if [ "${DOWNLOAD_LIST_IMAGES}" = "true"
+   # Parse it
+   echo ""
+   echo "---"
+-  printf "%-15s  %-10s  %-5s  %-7s  %-14s\n" "DIST" "RELEASE" "ARCH" "VARIANT" "BUILD"
++  printf "DIST\tRELEASE\tARCH\tVARIANT\tBUILD\n"
+   echo "---"
+   while IFS=';' read -r f1 f2 f3 f4 f5 f6; do
+     [ -n "${DOWNLOAD_DIST}" ] && [ "$f1" != "${DOWNLOAD_DIST}" ] && continue
+@@ -241,7 +241,7 @@ if [ "${DOWNLOAD_LIST_IMAGES}" = "true"
+     [ -n "${DOWNLOAD_VARIANT}" ] && [ "$f4" != "${DOWNLOAD_VARIANT}" ] && continue
+     [ -z "${f5}" ] || [ -z "${f6}" ] && continue
+ 
+-    printf "%-15s  %-10s  %-5s  %-7s  %-14s\n" "${f1}" "${f2}" "${f3}" "${f4}" "${f5}"
++    printf "%s\t%s\t%s\t%s\t%s\n" "${f1}" "${f2}" "${f3}" "${f4}" "${f5}"
+     unset f1 f2 f3 f4 f5 f6
+   done < "${DOWNLOAD_TEMP}/index"
+   echo "---"
-- 
2.49.0

